app/models/cohort_columns/last_homeless_visit.rb

```diff
@@ -14,17 +14,20 @@ def description
       'Date of last homeless service in ongoing enrollments'
     end
 
-    def value(cohort_client) # OK
-      cohort_client.client.processed_service_history&.last_homeless_visit&.
-        split(';')&.
-        map(&:strip)&.
-        sort do |a, b|
-          get_date(b) <=> get_date(a)
-        end&.join('; ')
-    end
+    def value(cohort_client, user) # OK
+      visible_project_ids = GrdaWarehouse::Hud::Project.viewable_by(user).pluck(:id)
```

Comment:
We should move this calculation into `User` so you can call `user.visible_project_ids` and cache it there.

```
def visible_project_ids 
  @visible_project_ids ||= GrdaWarehouse::Hud::Project.viewable_by(self).pluck(:id)
end
```

Reply:
✅